.github/workflows/ci.yml

```diff
@@ -0,0 +1,80 @@
+name: CI
+
+on:
+  push:
+  pull_request:
+  # Allow manual runs. It is necessary to run the workflow manually, when the "formatter" workflow applies any changes.
+  workflow_dispatch:
+
+jobs:
+  Processor_Partial:
+    if: github.event_name == 'push' || github.event.pull_request.head.repo.id != github.event.pull_request.base.repo.id
+    strategy:
+      matrix:
+        os: [ ubuntu-latest, windows-latest, macos-latest, macos-13 ] # macos-13 for x86_x64 arch
+        framework: [ net8.0 ]
+      fail-fast: false
+    runs-on: ${{ matrix.os }}
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+      - name: Setup .NET SDK v8.0.x
+        if: matrix.framework == 'net8.0'
+        uses: actions/setup-dotnet@v4
+        with:
+          dotnet-version: 8.0.x
```

Comment:
I think we should use the net9.0 sdk for building, and then the "matrix.framework" target for running the tests. So you would _always_ install the 9.0.x SDK, and _only_ install 8.0.x when `matrix.framework == 'net8.0'`. Remember to add the `net9.0` framework to the matrix, and then a special case of `net462` for Windows (via matrix includes).

Reply:
i think i did it. it is a little verbose, but that's generally it.

Reply:
i had to add net9.0 to project (not tests) so build suceeds even though it's not required in contrib project.

Reply:
this was because of the build step